src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala

```diff
@@ -281,16 +281,21 @@ private[io] object StageWriter {
     val writeTableState = new WriteTableState(conn)
 
     try {
+      val fullyQualifiedName = if (params.useFullyQualifiedName) {
```

Comment:
This is bad naming. If the param is false, fullyQualifiedName is actually not full qualified.
Just do
```
val tableName = if (params.useFullyQualifiedName) { 
  Utils.getFullyQualifiedName(params.table.get.name, params)
} else {
  params.table.get.name
}
```